standard-cluster-services/src/main/java/org/terracotta/entity/IEntityMessenger.java

```diff
@@ -47,4 +47,45 @@
    * @throws MessageCodecException The message could not be serialized.
    */
   void messageSelfAndDeferRetirement(EntityMessage originalMessageToDefer, EntityMessage newMessageToSchedule) throws MessageCodecException;
+
+  /**
+   * Requests that the given message be send to the entity instance after millisBeforeSend have elapsed.
+   * The message will be sent once, and may arrive late but won't arrive before millisBeforeSend have elapsed.
+   * The message will not arrive if the server restarts or the entity is destroyed before it was due.
+   * 
+   * @param message The new message to send.
+   * @param millisBeforeSend The minimum number of milliseconds which must pass before message should be delivered.
+   * @return A token to describe this delayed send which can be used to cancel it.
+   * @throws MessageCodecException The message could not be serialized.
+   */
+  ScheduledToken messageSelfAfterDelay(EntityMessage message, long millisBeforeSend) throws MessageCodecException;
```

Comment:
Wondering if this should be a Supplier<EntityMessage> so that a new version can be created on each iteration